admin/autoupdater/files/usr/sbin/autoupdater

```diff
@@ -257,21 +259,25 @@ local function autoupdate(mirror)
     return true
   end
 
+  autoupdater_util.run_dir('/usr/lib/autoupdater/download.d')
 
-  os.execute('sync; sysctl -w vm.drop_caches=3')
+  autoupdater_util.exec('sync')
+  autoupdater_util.exec('sysctl', '-w', 'vm.drop_caches=3')
   collectgarbage()
 
   local image = os.tmpname()
   if not fetch_firmware(mirror, manifest.filename, image) then
+    autoupdater_util.run_dir('/usr/lib/autoupdater/abort.d')
     return false
   end
 
-  local popen = io.popen(string.format("sha512sum '%s'", image))
+  local popen = io.popen(string.format("exec sha512sum '%s'", image))
   local checksum = popen:read('*l'):match('^%x+')
   popen:close()
   if checksum ~= manifest.checksum then
     io.stderr:write('Invalid image checksum!\n')
     os.remove(image)
+    autoupdater_util.run_dir('/usr/lib/autoupdater/abort.d')
```

Comment:
Indeed. I think making the path configurable is overkill, but having them defined at the top makes sense.